src/components/ProductsTable/ProductsTable.tsx

```diff
@@ -1,28 +1,54 @@
-import './ProductsTable.scss';
-import { ProductsTableRow } from 'components/ProductsTableRow/ProductsTableRow';
+import { ProductsTableRow } from './ProductsTableRow';
 import { useProducts } from 'hooks/useProducts';
 import { ReactElement } from 'react';
+import { useSearch } from 'components/Search/SearchContext';
+import { Search } from 'components/Search/Search';
+import { Button } from 'components/Buttons/Button';
+import { useProductFormContext } from 'components/ProductForm/ProductFormContext';
+import { ProductDto } from 'dtos/ProductDto';
+import { LoaderOverlay } from 'components/LoaderOverlay/LoaderOverlay';
 
 export const ProductsTable = (): ReactElement => {
   const { products } = useProducts();
+  const formatting = ' px-4 py-2 font-normal ';
+  const searchString = useSearch().searchString;
+  const displayProducts: ProductDto[] =
+    products.data?.filter((product) => !product.retired && product.name?.toLocaleLowerCase().includes(searchString.toLocaleLowerCase())) || [];
+  const openModal = useProductFormContext().startNew;
 
   return (
-    <table className="products_table">
-      <thead>
-        <tr>
-          <th className="text-left">ID</th>
-          <th className="text-left">Name</th>
-          <th className="products_table__hide_mobile text-right">Price</th>
-          <th className="products_table__hide_mobile text-left">Description</th>
-          <th className="products_table__hide_mobile text-left">Categories</th>
-          <th />
-        </tr>
-      </thead>
-      <tbody>
-        {products.data?.map((product) => (
-          <ProductsTableRow product={product} key={product.id} />
-        ))}
-      </tbody>
-    </table>
+    <>
+      <div className="w-full overflow-auto">
+        {displayProducts.length ? (
+          <table className="w-full table-auto divide-y">
+            <thead className="sticky top-0 w-full bg-white/90">
+              <tr>
+                <th className={'text-left ' + formatting}>ID</th>
+                <th className={'w-full text-left ' + formatting}>Name</th>
+                <th className={'text-center ' + formatting}>Category</th>
+                <th className={'text-center ' + formatting}>Stock</th>
+                <th className={'text-center ' + formatting}>Price</th>
+                <th className={'whitespace-nowrap text-center' + formatting}>No. Sold</th>
+                <th />
+              </tr>
+            </thead>
+            <tbody className="font-poppins font-200 divide-y text-sm">
+              {displayProducts.map((product) => (
+                <ProductsTableRow formatting={formatting} product={product} key={product.id} />
+              ))}
+            </tbody>
+          </table>
+        ) : (
```

Comment:
If there are no products we still want to be able to see the table headers! Remove this conditional for other tables too if it is there